src/components/button/Button.tsx

```diff
@@ -53,22 +57,51 @@ class Button extends React.Component<ButtonProps> {
     };
 
     render() {
-        const { children, className, isDisabled, isLoading, isSelected, setRef, type, showRadar, ...rest } = this.props;
+        const {
+            children,
+            className,
+            icon,
+            isDisabled,
+            isLoading,
+            isSelected,
+            setRef,
+            size,
+            type,
+            showRadar,
+            ...rest
+        } = this.props;
         const buttonProps: Record<string, boolean> = omit(rest, ['onClick']);
         if (isDisabled) {
             buttonProps['aria-disabled'] = true;
         }
 
+        const hasIcon = !!icon;
+        const hasText = !!children;
```

Comment:
Should we be making the assumption that `children` will only contain text? If a consumer passes in an icon on the left, for example?